library/src/otx/data/entity/sample.py

```diff
@@ -0,0 +1,298 @@
+            ori_shape=shape,
+        )
+
+
+@register_pytree_node
+class DetectionSample(OTXSample):
+    """DetectionSample is a base class for OTX detection items."""
+
+    image: tv_tensors.Image | np.ndarray | torch.Tensor = image_field(
+        dtype=pl.UInt8(), format="BGR", channels_first=True
+    )
+    label: torch.Tensor = label_field(pl.Int32(), is_list=True)
+    bboxes: np.ndarray | tv_tensors.BoundingBoxes = bbox_field(dtype=pl.Float32())
+    dm_image_info: DmImageInfo = image_info_field()
+
+    def __post_init__(self) -> None:
+        shape = (self.dm_image_info.height, self.dm_image_info.width)
+
+        # Convert bboxes to tv_tensors format
+        if isinstance(self.bboxes, np.ndarray):
+            self.bboxes = tv_tensors.BoundingBoxes(
+                self.bboxes,
+                format=tv_tensors.BoundingBoxFormat.XYXY,
+                canvas_size=shape,
+                dtype=torch.float32,
+            )
+
+        self.img_info = ImageInfo(
+            img_idx=0,
+            img_shape=shape,
+            ori_shape=shape,
+        )
+
+
+@register_pytree_node
+class SegmentationSample(OTXSample):
+    """OTXDataItemSample is a base class for OTX data items."""
+
+    subset: Subset = subset_field()
+    image: np.ndarray | tv_tensors.Image | torch.Tensor = image_field(dtype=pl.UInt8(), channels_first=True)
+    masks: tv_tensors.Mask = mask_field(dtype=pl.UInt8(), channels_first=True, has_channels_dim=True)
+    dm_image_info: DmImageInfo = image_info_field()
+
+    def __post_init__(self) -> None:
+        shape = (self.dm_image_info.height, self.dm_image_info.width)
+        self.img_info = ImageInfo(
+            img_idx=0,
+            img_shape=shape,
+            ori_shape=shape,
+        )
+
+
+@register_pytree_node
+class InstanceSegmentationSample(OTXSample):
+    """OTXSample for instance segmentation tasks."""
+
+    subset: Subset = subset_field()
+    image: tv_tensors.Image | np.ndarray | torch.Tensor = image_field(dtype=pl.UInt8(), channels_first=True)
+    bboxes: np.ndarray | tv_tensors.BoundingBoxes = bbox_field(dtype=pl.Float32())
+    label: torch.Tensor = label_field(is_list=True)
+    polygons: np.ndarray = polygon_field(dtype=pl.Float32())  # Ragged array of (Npoly, 2) arrays
+    dm_image_info: DmImageInfo = image_info_field()
+
+    def __post_init__(self) -> None:
+        shape = (self.dm_image_info.height, self.dm_image_info.width)
+
+        # Convert bboxes to tv_tensors format
+        if isinstance(self.bboxes, np.ndarray):
+            self.bboxes = tv_tensors.BoundingBoxes(
+                self.bboxes,
+                format=tv_tensors.BoundingBoxFormat.XYXY,
+                canvas_size=shape,
+                dtype=torch.float32,
+            )
+
+        self.img_info = ImageInfo(
+            img_idx=0,
+            img_shape=shape,
+            ori_shape=shape,
+        )
+
+
+@register_pytree_node
+class InstanceSegmentationSampleWithMask(OTXSample):
```

Comment:
1) Can't we combine InstanceSegmentationSample and InstanceSegmentationSampleWithMask? The difference in mask field only. 
2) Why do we still include polygons in InstanceSegmentationSampleWithMask?